examples/research_projects/zero-shot-distillation/distill_classifier.py

```diff
@@ -0,0 +1,320 @@
+class names.
+"""
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class TeacherModelArguments:
+    teacher_name_or_path: Optional[str] = field(
+        default="roberta-large-mnli", metadata={"help": "The NLI/zero-shot teacher model to be distilled."}
+    )
+    hypothesis_template: Optional[str] = field(
+        default="This example is {}.",
+        metadata={
+            "help": (
+                "Template used to turn class names into mock hypotheses for teacher NLI model. Must include {{}}"
+                "where class name is inserted."
+            )
+        },
+    )
+    teacher_batch_size: Optional[int] = field(
+        default=32, metadata={"help": "Batch size for generating teacher predictions."}
+    )
+    multi_class: Optional[bool] = field(
+        default=False,
+        metadata={
+            "help": (
+                "Allow multiple classes to be true rather than forcing them to sum to 1 (sometimes called"
+                "multi-class multi-label classification)."
+            )
+        },
+    )
+    temperature: Optional[float] = field(
+        default=1.0, metadata={"help": "Temperature applied to teacher softmax for distillation."}
+    )
+
+
+@dataclass
+class StudentModelArguments:
+    student_name_or_path: Optional[str] = field(
+        default="distilbert-base-uncased", metadata={"help": "The NLI/zero-shot teacher model to be distilled."}
+    )
+
+
+@dataclass
+class DataTrainingArguments:
+    data_file: str = field(metadata={"help": "Text file with one unlabeled instance per line."})
+    class_names_file: str = field(metadata={"help": "Text file with one class name per line."})
+    use_fast_tokenizer: bool = field(
+        default=True,
+        metadata={"help": "Whether to use one of the fast tokenizer (backed by the Rust tokenizers library) or not."},
+    )
+
+
+@dataclass
+class DistillTrainingArguments(TrainingArguments):
+    output_dir: Optional[str] = field(
+        default=None,
+        metadata={"help": "The output directory where the model predictions and checkpoints will be written."},
+    )
+    per_device_train_batch_size: int = field(
+        default=32, metadata={"help": "Batch size per GPU/TPU core/CPU for training."}
+    )
+    per_device_eval_batch_size: int = field(
+        default=128, metadata={"help": "Batch size per GPU/TPU core/CPU for evaluation."}
+    )
+    num_train_epochs: float = field(default=1.0, metadata={"help": "Total number of training epochs to perform."})
+    do_train: bool = field(default=True, metadata={"help": "Whether to run training of student model."})
+    do_eval: bool = field(
+        default=True,
+        metadata={
+            "help": (
+                "Whether to evaluate the agreement of the final student predictions and the teacher predictions"
+                "after training."
+            )
+        },
+    )
```

Comment:
All those arguments are already in `TrainingArguments`, so this class is the same as `TrainingArguments`. Not sure what you were trying to achieve.

Reply:
It subclasses `TrainingArguments` and overrides the default values.